internal/controller/backup_controller.go

```diff
@@ -287,7 +287,10 @@ func (r *BackupReconciler) isValidBackupRunning(
 		return false, fmt.Errorf("unknown.spec.target received: %s", backup.Spec.Target)
 	}
 
-	containerIsNotRestarted := backup.Status.InstanceID.ContainerID == pod.Status.ContainerStatuses[0].ContainerID
+	var containerIsNotRestarted bool
+	if len(pod.Status.ContainerStatuses) > 0 && backup.Status.InstanceID != nil {
```

Comment:
@jsilvela now that you introduced the utility function, you could invoke it also here, replacing with

`if !utils.IsPodReady(pod) && backup.Status.InstanceID != nil {`